web/src/__tests__/score-configs.servertest.ts

```diff
@@ -0,0 +1,115 @@
+/** @jest-environment node */
+
+import { makeAPICall, pruneDatabase } from "@/src/__tests__/test-utils";
+import { v4 as uuidv4 } from "uuid";
+import { ScoreDataType, prisma } from "@langfuse/shared/src/db";
+import { type CastedConfig } from "@langfuse/shared";
+
+const CONFIG_ID_ONE = uuidv4();
+const CONFIG_ID_TWO = uuidv4();
+const CONFIG_ID_THREE = uuidv4();
+
+describe("/api/public/score-configs API Endpoint", () => {
+  let should_prune_db = true;
+  beforeAll(
+    async () => {
+      if (should_prune_db) await pruneDatabase();
+
+      const configOne = [
+        {
+          id: CONFIG_ID_ONE,
+          projectId: "7a88fb47-b4e2-43b8-a06c-a5ce950dc53a",
+          name: "Test Boolean Config",
+          description: "Test Description",
+          dataType: ScoreDataType.BOOLEAN,
+          categories: [
+            { label: "False", value: 0 },
+            { label: "True", value: 1 },
+          ],
+        },
+      ];
+      const configTwo = [
+        {
+          id: CONFIG_ID_TWO,
+          projectId: "7a88fb47-b4e2-43b8-a06c-a5ce950dc53a",
+          name: "Test Numeric Config",
+          description: "Test Description",
+          dataType: ScoreDataType.NUMERIC,
+          minValue: 0,
+        },
+      ];
+
+      const configThree = [
+        {
+          id: CONFIG_ID_THREE,
+          projectId: "7a88fb47-b4e2-43b8-a06c-a5ce950dc53a",
+          name: "Test Categorical Config",
+          description: "Test Description",
+          dataType: ScoreDataType.CATEGORICAL,
+          categories: [
+            { label: "A", value: 0 },
+            { label: "B", value: 1 },
+            { label: "C", value: 2 },
+          ],
+        },
+      ];
+
+      await prisma.scoreConfig.createMany({
+        data: [...configOne, ...configTwo, ...configThree],
+      });
+    },
+    afterAll(async () => {
+      if (should_prune_db) await pruneDatabase();
+    }),
+  );
+
+  it("should GET a score config", async () => {
+    const configId = CONFIG_ID_ONE;
+
+    const getScoreConfig = await makeAPICall<{
+      id: string;
+    }>("GET", `/api/public/score-configs/${configId}`);
+
+    expect(getScoreConfig.status).toBe(200);
+    expect(getScoreConfig.body).toMatchObject({
+      id: configId,
+    });
+  });
+
+  it("test invalid config id input", async () => {
+    const configId = "invalid-config-id";
+
+    const getScoreConfig = await makeAPICall<{
+      message: string;
+    }>("GET", `/api/public/score-configs/${configId}`);
+
+    expect(getScoreConfig.status).toBe(404);
+    expect(getScoreConfig.body).toMatchObject({
+      message: "Score config not found within authorized project",
+    });
+  });
+
+  it("should GET all score configs", async () => {
+    const fetchedConfigs = await makeAPICall<{
+      data: CastedConfig[];
```

Comment:
can we be certain if we do not verify? unsure if we should verify here but I would not cast if we do not check here

Reply:
I chose to verify and then cast to ensure data consistency. I don't see why we should allow any other config.categories formats at the moment. We can always become less permissive in the future if it's needed.